Home: Masthead v1

Diabeticodes opened the pull request:
# Why
In generality a new branch needed made from development and synced to include initial changes from the 'mike-dev' branch. This also included the list of changes requested in the original PR...all of which have been added in the following section below.

## What
- Import statements reorganized
- Logo changed to rectangle with text and bg color instead of image
- Margin adjusted between ImageMasthead and LatestPosts component
- Removed About and Blog page since they've been re-structured in the project.

## Testing Notes (if applicable)
There is a certain smaller resolution in which the logo isn't completely centered. May need to double check to make sure it's not a common device screen size and more of a 'gray area' in which the site wouldn't normally be viewed.

## Checklist

- [X] Looks good on large screens
- [X] Looks good on mobile screens

Commit:
added photos and component

optimized 3 images for home page along with component template

Commit:
changed component name

Commit:
changed component name

Commit:
Images and logo added to homepage

Commit:
Images and logo added to homepage

Commit:
Added links to images

Commit:
About content added

Simplified Logo component,
Updated Masthead

Commit:
About content added

Simplified Logo component,
Updated Masthead

giantrotta24 committed:
Co-authored-by: Michael <michael.mastrucci@gmail.com>

giantrotta24 committed:
add compressed images to masthead

giantrotta24 committed:
add compressed images to masthead

Commit:
Compressed images

giantrotta24 committed:
update logo

giantrotta24 committed:
test

Commit:
added responsive styles for image layering in masthead display

Commit:
added responsive styles for image layering in masthead display

Commit:
finalized responsive styles for logo and image alignment and display

Commit:
fixed class errors

Commit:
logo updated

Commit:
camel casing fix

replaced logo title with camel casing to remain consistent

Commit:
Merge branch 'mike-dev' into michael

Commit:
Syncing changes from dev to here

Commit:
Merge branch 'mike-dev' into michael

Commit:
Updated to include icons

Commit:
Removed pages

Based on Github comments, About and Blog pages have been removed and the import statements have been reorganized for project consistency

giantrotta24 reviewed (commented):
Not a big deal because it's just the two of us but just some pointers around PRs. Again, your description overall following the template is fantastic but you did not name your PR, so it took the branch name as default which wouldnt be super helpful to someone just scanning through PRs quickly.

giantrotta24 reviewed (commented):
Another pointer, you are merging to master which does not follow our merging structure. Your development branches should be making their way to the `development` branch.

And finally, if you felt it was better to start fresh with a new branch/PR then it is always good practice to close the previous PR with a relevant comment.

Commit:
added contact form to home page

Commit:
Grab user input

Added type guards to ensure input doesnt store a null value before accessing the value property of each input field/text area

giantrotta24 committed:
Merge branch 'development' of https://github.com/giantrotta24/mich-site into michael

giantrotta24 commented on src/pages/index.tsx:
```diff
@@ -1,17 +1,15 @@
-import ContactMe from '../components/Home/ContactMe';
+import HomeMasthead from '../components/HomeMasthead';
+import LatestPosts from '../components/Home/LatestPosts/index';
+import Contact from '../components/Home/ContactMe'
```
The about me section you are removing here is the one we want to keep, that is the one I merged in.

Diabeticodes replied:
Added

giantrotta24 commented on yarn.lock:
```diff
@@ -0,0 +1,2013 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
looks like you ran yarn instead of npm, you'll want to make sure this file is not committed to git history

giantrotta24 commented on src/components/Home/ContactMe/ContactMeForm.tsx:
```diff
@@ -5,7 +5,17 @@ const ContactMeForm: React.FC = () => {
 
   const onSubmit = (data: any) => {
-    console.log(data);
+    //Attempt to store input data with type assertion
```
Can we revert this change and move this to a new branch?

Diabeticodes replied:
reverted

giantrotta24 commented on src/components/common/Logo.tsx:
```diff
@@ -0,0 +1,19 @@
+const Logo = () => {
+  return (
+    <div className="relative bottom-36 lg:left-96 md:left-0">
+      <Image
```
Logo looks great! Can we adjust a few things? There is a typo, should be "Toxin free" not "Toxic", can we adjust the border radius so it's not curved (nothing in our design is curved so we want to keep that standard), and not sure if this is using our default font, which should be `font-family: ui-serif, Georgia, Cambria, "Times New Roman", Times, serif;`

giantrotta24 reviewed (changes requested):
Great work, almost there! just a few things we need to change to keep our git history clean and some minor tweaks to the logo. 

Also, if you don't want to keep messing with the logo in photoshop im pretty sure you could make the same thing happen with code since it's just a square with text ;) but that's definitely not necessary, your call!

giantrotta24 committed:
switch to yarn

Commit:
About Me

Added component back in

Commit:
reverted form functionality

will add to new branch

giantrotta24 committed:
fix conflicts, merge rebase

giantrotta24 committed:
fix everything

giantrotta24 committed:
remove images, fix conflict

giantrotta24 committed:
Merge branch 'development' into michael

giantrotta24 committed:
yarn install

giantrotta24 committed:
remove files

Commit:
Swapped Logo

Image changed for text with bg color

giantrotta24 reviewed (approved):
Great work! Lets merge this during our next pairing session!
